src/components/ServerPaginationGrid.jsx

```diff
@@ -0,0 +1,77 @@
+import { useContext, useMemo, useState } from "react";
+import useSWR from "swr";
+import { DataGrid } from "@mui/x-data-grid";
+import LinearProgress from "@mui/material/LinearProgress";
+
+import { useMeasurements } from "../lib/hooks/swr-extensions";
+import { PreferenceContext } from "../pages/_app";
+
+const endpoint = "http://localhost:3000/api/v2/data?";
+const urlWithParams = (baseUrl, params) => baseUrl + new URLSearchParams(params);
+const fetcher = async (url) => {
+  const response = await fetch(url);
+  return await response.json();
+};
+
+const ServerPaginationGrid = () => {
+  /* Define pagination options, which can be modified in the grid */
+  const [page, setPage] = useState(0);
+  const [pageSize, setPageSize] = useState(20);
+
+  /* Get correct url for fetching the filtered data */
+  const { preferences } = useContext(PreferenceContext);
+  const url = useMemo(() => urlWithParams(endpoint, {
+    temperature_unit: preferences.temperature_unit.symbol,
+    conductivity_unit: preferences.conductivity_unit.symbol,
+    location_name: preferences.location.symbol,
+    page: page + 1, /* mui grid starts indexing at 0, api at 1 */
+    page_size: pageSize,
+  }), [preferences, page, pageSize]);
+
+  const { measurements, rowCount, isLoading } = useMeasurements(url, fetcher);
+
+  const gridColumns = useMemo(() => [
```

Comment:
prop `editable` for [GridColDef](https://mui.com/api/data-grid/grid-col-def/) is false by default